sql/internal/spectest/spectest.go

```diff
@@ -0,0 +1,62 @@
+package spectest
+
+import (
+	"reflect"
+	"testing"
+
+	"ariga.io/atlas/schema/schemaspec"
+	"ariga.io/atlas/sql/internal/specutil"
+	"ariga.io/atlas/sql/schema"
+	"github.com/stretchr/testify/require"
+)
+
+// RegistrySanityTest runs a sanity for a TypeRegistry, generated a dummy *schemaspec.Type
+// then converting it to a schema.Type and back to a *schemaspec.Type.
+func RegistrySanityTest(t *testing.T, registry *specutil.TypeRegistry, parser func(string) (schema.Type, error), skip []string) {
+	for _, ts := range registry.Specs() {
+		if contains(ts.Name, skip) {
+			continue
+		}
+		t.Run(ts.Name, func(t *testing.T) {
+			spec := dummyType(t, ts)
+			styp, err := registry.Type(spec, nil, parser)
+			require.NoError(t, err)
+			require.NoErrorf(t, err, "failed formatting: %styp", err)
+			convert, err := registry.Convert(styp)
+			require.NoError(t, err)
+			after, err := registry.Type(convert, nil, parser)
+			require.NoError(t, err)
+			require.EqualValues(t, styp, after)
+		})
+	}
+}
+
+func contains(needle string, haystack []string) bool {
+	for _, item := range haystack {
+		if item == needle {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
What are these var names?
```suggestion
func contains(s string, l []string) bool {
	for i := range l {
		if s == l[i] {
			return true
		}
	}
	return false
}
```